mlflow/genai/labeling/stores.py

```diff
@@ -0,0 +1,487 @@
+            enable_multi_turn_chat: Whether to enable multi-turn chat labeling for the session.
+            custom_inputs: Optional. Custom inputs to be used in the session.
+            experiment_id: The experiment ID. If None, uses the currently active experiment.
+
+        Returns:
+            LabelingSession: The created labeling session.
+        """
+
+    @abstractmethod
+    def delete_labeling_session(self, labeling_session: LabelingSession) -> None:
+        """
+        Delete a labeling session.
+
+        Args:
+            labeling_session: The labeling session to delete.
+        """
+
+    @abstractmethod
+    def get_label_schema(self, name: str) -> LabelSchema:
+        """
+        Get a label schema by name.
+
+        Args:
+            name: The name of the label schema.
+
+        Returns:
+            LabelSchema: The label schema.
+
+        Raises:
+            mlflow.MlflowException: If label schema is not found.
+        """
+
+    @abstractmethod
+    def create_label_schema(
+        self,
+        name: str,
+        *,
+        type: str,
+        title: str,
+        input: Any,
+        instruction: str | None = None,
+        enable_comment: bool = False,
+        overwrite: bool = False,
+    ) -> LabelSchema:
+        """
+        Create a new label schema.
+
+        Args:
+            name: The name of the label schema. Must be unique across the review app.
+            type: The type of the label schema. Either "feedback" or "expectation".
+            title: The title of the label schema shown to stakeholders.
+            input: The input type of the label schema.
+            instruction: Optional. The instruction shown to stakeholders.
+            enable_comment: Optional. Whether to enable comments for the label schema.
+            overwrite: Optional. Whether to overwrite the existing label schema with the same name.
+
+        Returns:
+            LabelSchema: The created label schema.
+        """
+
+    @abstractmethod
+    def delete_label_schema(self, name: str) -> None:
+        """
+        Delete a label schema.
+
+        Args:
+            name: The name of the label schema to delete.
+        """
+
+    @abstractmethod
+    def add_dataset_to_session(
+        self,
+        labeling_session: LabelingSession,
+        dataset_name: str,
+        record_ids: list[str] | None = None,
+    ) -> LabelingSession:
+        """
+        Add a dataset to a labeling session.
+
+        Args:
+            labeling_session: The labeling session to add the dataset to.
+            dataset_name: The name of the dataset.
+            record_ids: Optional. The individual record ids to be added to the session.
+
+        Returns:
+            LabelingSession: The updated labeling session.
+        """
+
+    @abstractmethod
+    def add_traces_to_session(
+        self,
+        labeling_session: LabelingSession,
+        traces: Union[Iterable[Trace], Iterable[str], "pd.DataFrame"],
```

Comment:
As a follow-up, this can likely just be simplified to a set of Trace objects or trace IDs at the store level, but out of scope for this PR.